script/check_copyright_year.sh

```diff
@@ -2,18 +2,70 @@
 # Copyright (c) Advanced Micro Devices, Inc., or its affiliates.
 # SPDX-License-Identifier: MIT
 
+# This script checks if files have the correct copyright header template.
+# It supports .hpp, .cpp, .inc, .py, .sh, and .cmake files.
+#
+# Usage: ./check_copyright_year.sh <file1> <file2> ...
 
-current_year=$(date +%Y)
 exit_code=0
 
-for file in $@; do
-    if grep -q "Copyright (c)" $file
-    then
-        if ! grep -q "Copyright (c).*$current_year" $file
-        then
-            echo "ERROR: File $file has a copyright notice without the current year ($current_year)."
-            exit_code=1
-        fi
+# Expected copyright header lines (without comment characters)
+COPYRIGHT_LINE="Copyright (c) Advanced Micro Devices, Inc., or its affiliates."
+SPDX_LINE="SPDX-License-Identifier: MIT"
+
+check_file() {
+    local file=$1
+    local ext="${file##*.}"
+    local comment_char
+
+    # Determine comment character based on file extension
+    case "$ext" in
+        cpp|hpp|inc)
+            comment_char="//"
+            ;;
+        py|sh|cmake)
+            comment_char="#"
+            ;;
+        *)
+            # If the filename is *.cmake or CMakeLists.txt, use #
+            if [[ "$file" == *CMakeLists.txt ]] || [[ "$file" == *.cmake ]]; then
+                comment_char="#"
+            else
+                # Skip files with unsupported extensions
+                return 0
+            fi
+            ;;
+    esac
```

Comment:
Can you simplify this?

```suggestion
        py|sh|cmake|CMakeLists.txt)
            comment_char="#"
            ;;
        *)
            # Skip files with unsupported extensions
            return 0
            ;;
    esac
```

Reply:
Thank you and fixed. https://github.com/ROCm/composable_kernel/pull/3302/commits/e72dd0fbba8575038f0c513da9d9f7c97718e7d7